src/lib.rs

```diff
@@ -636,6 +649,32 @@ storage_types! {
     }
 }
 
+storage_types! {
+    types: Complex;
+    impl crate::Conversion<V> for V {
+        type T = VV;
+
+        #[inline(always)]
+        fn conversion(&self) -> Self::T {
+            // Conversion factor is the norm of the number.
+            // Scaling with length again yields the same number.
+            self.norm()
+        }
+    }
+    impl crate::ConversionFactor<V> for VV {
+        #[inline(always)]
+        fn powi(self, e: i32) -> Self {
+            self.powi(e)
+        }
+        #[inline(always)]
+        fn value(self) -> V {
+            // Conversion by scaling (multiplication with only real number)
+            // Scaling a normalized number yields the original number again.
+            V::new(self, 0.0)
+        }
+    }
+}
```

Comment:
* Some minor formatting.
 *  `fn constant` is needed to ensure zero-cost code generation. See documentation on the trait for details.

```suggestion
storage_types! {
    types: Complex;

    impl crate::Conversion<V> for V {
        type T = VV;

        #[inline(always)]
        fn constant(op: crate::ConstantOp) -> Self::T {
            match op {
                crate::ConstantOp::Add => -<Self::T as crate::num::Zero>::zero(),
                crate::ConstantOp::Sub => <Self::T as crate::num::Zero>::zero(),
            }
        }

        #[inline(always)]
        fn conversion(&self) -> Self::T {
            // Conversion factor is the norm of the number. Scaling with length again yields the
            // same number.
            self.norm()
        }
    }

    impl crate::ConversionFactor<V> for VV {
        #[inline(always)]
        fn powi(self, e: i32) -> Self {
            self.powi(e)
        }

        #[inline(always)]
        fn value(self) -> V {
            // Conversion by scaling (multiplication with only real number). Scaling a normalized
            // number yields the original number again.
            V::new(self, 0.0)
        }
    }
}
```